src/app/app.component.ts

```diff
@@ -0,0 +1,57 @@
+import { Component, OnInit } from '@angular/core';
+import { CoffeeShopsService } from './services/coffeeShops.service';
+import {  retry, switchMap } from 'rxjs';
+import { CoffeeShopModel } from './models/coffeeShop.model';
+import { FormControl, FormGroup, Validators } from '@angular/forms';
+
+@Component({
+  selector: 'app-root',
+  templateUrl: './app.component.html',
+  styleUrls: ['./app.component.scss']
+})
+export class AppComponent implements OnInit {
+  coffeeShops = new Array<CoffeeShopModel>();
+  coordinatesFormControl = new FormGroup({
+    xCoordinate: new FormControl([], Validators.required),
+    yCoordinate: new FormControl([], Validators.required)
+  });
+
+  constructor(private coffeeShop: CoffeeShopsService) {
+  }
+
+  ngOnInit(): void {
+    this.getCoffeeShops();
+  }
+
+  getCoffeeShops(): void {
+    this.coffeeShop.getToken().pipe(
+      switchMap((result: any) => {
+        return this.coffeeShop.getCoffeeShops(result.token);
+      }), retry(5)).subscribe(result => this.coffeeShops = result);
+
+  }
+
+  getNearbyCS(): void {
+    for (let coffeeShop of this.coffeeShops) {
```

Comment:
Isn't there a more high level function in JS for transforming values of an array into different values?

Reply:
Yes, the map, but I didn't know the outcome when I started writing the logic.